pkg/operator/configobservation/apiserver/observe_admission_plugins.go

```diff
@@ -0,0 +1,92 @@
+package apiserver
+
+import (
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type pluginCheckerFunc func(listers configobservation.Listers) (enabled, disabled []string, err error)
+
+var (
+	enableAdmissionPluginsPath  = []string{"apiServerArguments", "enable-admission-plugins"}
+	disableAdmissionPluginsPath = []string{"apiServerArguments", "disable-admission-plugins"}
+
+	pluginCheckers = []pluginCheckerFunc{
+		roleBindingRestrictionPluginChecker,
+	}
+)
+
+// ObserveAdmissionPlugins manages the apiServerArguments.enable-admission-plugins and
+// apiServerArguments.disable-admission-plugins fields of the configuration. It defines a list of
+// plugin checkers which check the state of specific plugins, and add them to the enabled or disabled
+// list as required. This observer will overwrite any pre-existing values of the two fields in the existingConfig.
+func ObserveAdmissionPlugins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]any) (ret map[string]any, _ []error) {
+	defer func() {
+		ret = configobserver.Pruned(ret, enableAdmissionPluginsPath, disableAdmissionPluginsPath)
+	}()
+
+	if len(pluginCheckers) == 0 {
+		return existingConfig, nil
+	}
+
+	listers := genericListers.(configobservation.Listers)
+	enabledSet := sets.NewString()
+	disabledSet := sets.NewString()
```

Comment:
There are now generic versions of these. Just be aware of the sorting the values if needed (e.g. `slices.Sorted(slices.Values(enabledSet.UnsortedList()))`)

Reply:
Thanks, good to know! Updated to use the new generic versions of the sets.